core/src/main/scala/sbt/librarymanagement/ResolverExtra.scala

```diff
@@ -153,12 +155,20 @@ private[librarymanagement] abstract class ResolverFunctions {
     url("sbt-plugin-" + status, new URL(SbtRepositoryRoot + "/sbt-plugin-" + status + "/"))(
       ivyStylePatterns
     )
+  @deprecated("Use sonatypeRepos instead", "1.7.0")
   def sonatypeRepo(status: String) =
     MavenRepository(
       "sonatype-" + status,
       if (status == "releases") SonatypeReleasesRepository
       else SonatypeRepositoryRoot + "/" + status
     )
+  private def sonatypeS01Repo(status: String) =
+    MavenRepository(
+      "sonatype-s01-" + status,
+      SonatypeS01RepositoryRoot + "/" + status
+    )
+  def sonatypeRepos(status: String) =
+    Vector(sonatypeRepo(status): @nowarn("cat=deprecation"), sonatypeS01Repo(status))
```

Comment:
The difference between `sonatypeRepo(...)` and `sonatypeRepos(...)` is a bit subtle, so I'd suggest `allSonatypeOssRepos(...)`.

Reply:
Good point. I like `sonatypeOssRepos`: starting with `sonatype` is good for discoverability, and they are all `*oss.sonatype.org` so I suppose there could be other non-OSS repos.